test/FastTests/Utils/TimeParsing.cs

```diff
@@ -151,7 +154,130 @@ public void TimeOnly(string date, int hh, int mm, int ss, int ms)
+        {
+            var expected = DateTime.ParseExact(dt, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture,
+                DateTimeStyles.RoundtripKind);
+
+            var formatsToRead = new Dictionary<string, DateTimeKind>
+            {
+                {DefaultFormat.DateTimeFormatsToRead[0], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[3], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[5], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[6], DateTimeKind.Utc},
+            };
+
+            foreach (var dateTimeFormat in formatsToRead)
+            {
+                string tested = expected.ToString(dateTimeFormat.Key);
+
+                var expectedAfterFormatting = DateTime.ParseExact(tested, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture,
+                    DateTimeStyles.RoundtripKind);
+
+                var bytes = Encoding.UTF8.GetBytes(tested);
+                fixed (byte* buffer = bytes)
+                {
+                    Assert.Equal(LazyStringParser.Result.DateTime,
+                        LazyStringParser.TryParseDateTime(buffer, bytes.Length, out DateTime time, out _, properlyParseThreeDigitsMilliseconds: true));
+                    Assert.Equal(expectedAfterFormatting, time);
+                }
+            }
+        }
+
+        [RavenTheory(RavenTestCategory.Core)]
+        [InlineDataWithRandomSeed]
+        public void CanParseValidRandomDate(int seed)
+        {
+            var r = new Random(seed);
+
+            var dt = GetRandomDate(r);
+
+            var formatsToRead = new Dictionary<string, DateTimeKind>
+            {
+                {DefaultFormat.DateTimeFormatsToRead[0], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[1], DateTimeKind.Unspecified},
+                {DefaultFormat.DateTimeFormatsToRead[2], DateTimeKind.Local},
+                {DefaultFormat.DateTimeFormatsToRead[3], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[4], DateTimeKind.Unspecified},
+                {DefaultFormat.DateTimeFormatsToRead[5], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[6], DateTimeKind.Utc},
+            };
+            
+            Assert.Equal(formatsToRead.Count, DefaultFormat.DateTimeFormatsToRead.Length);
+
+            foreach (var dateTimeFormat in formatsToRead)
+            {
+                string tested = dt.ToString(dateTimeFormat.Key);
+
+                var bytes = Encoding.UTF8.GetBytes(tested);
+                fixed (byte* buffer = bytes)
+                {
+                    var parseResult = LazyStringParser.TryParseDateTime(buffer, bytes.Length, out var dateTime, out var dateTimeOffset, properlyParseThreeDigitsMilliseconds: true);
+                    
+                    Assert.NotEqual(LazyStringParser.Result.Failed, parseResult);
```

Comment:
I would add the 'tested' value to the message here

Reply:
Changed to Assert.True and added a message